tls/s2n_key_update.c

```diff
@@ -110,6 +110,9 @@ int s2n_check_record_limit(struct s2n_connection *conn, struct s2n_blob *sequenc
     uint64_t output = 0;
     POSIX_GUARD(s2n_sequence_number_to_uint64(sequence_number, &output));
 
+    /* Safety: `s2n_increment_sequence_number()` will return an error and terminate the connection if
+     * the sequence number reaches UINT64_MAX. This means that we can assume `output < UINT64_MAX`
+     * is always true. This prevents `output` from overflowing. */
```

Comment:
Why not just add `POSIX_ENSURE_LT(output, UINT64_MAX);` or `if (output < UINT64_MAX && output + 1...`?

Reply:
or just `if (output >= conn->secure.cipher_suite->record_alg->encryption_limit)`

Reply:
Initially I was hesitant to change the behavior of this function but after chatting with team `if (output >= conn->secure.cipher_suite->record_alg->encryption_limit)` should be fine

Reply:
Scratch that, `conn->secure.cipher_suite->record_alg->encryption_limit` is allowed to be 0, same with `output`

Reply:
If `output` is 0, the condition should always evaluate to false. Thus, the following should be sufficient to avoid overflow and preserve behavior, right?
```c
if (output != 0 && output >= conn->secure.cipher_suite->record_alg->encryption_limit)
```
Take a look at this [commit](https://github.com/feliperodri/s2n/commit/c8167d2a617a0263c2f08520a5e2a07eccbd68fc). It has a CBMC proof of memory safety for this function for any given input using the suggestion above.

Reply:
Thanks for the example, I'll pull it in. There was some other issues that cropped up with this change that I still need to address, so it may be a bit